vault/resource_gcp_auth_backend_test.go

```diff
@@ -113,3 +174,41 @@ resource "vault_gcp_auth_backend" "test" {
 }
 `, credentials, path)
 }
+
+func testGCPAuthBackendConfig_update(path, credentials string) string {
+	return fmt.Sprintf(`
+variable "json_credentials" {
+  type = string
+  default = %q
+}
+
+resource "vault_gcp_auth_backend" "test" {
+  path                          = %q
+  credentials                   = var.json_credentials
+  custom_endpoint {
+    api     = "www.googleapis.com"
+    iam     = "iam.googleapis.com"
+    crm     = "cloudresourcemanager.googleapis.com"
+    compute = "compute.googleapis.com"
+  }
+}
+`, credentials, path)
+}
+
+func testGCPAuthBackendConfig_update_partial(path, credentials string) string {
+	return fmt.Sprintf(`
+variable "json_credentials" {
+  type = string
+  default = %q
```

Comment:
Typically I like to use `"%s"` for the quoting instead of using `%q` since I can  pipe the selected block e.g. `!terraform fmt -` with Vim. The latter tends to break the pipe.

Reply:
This was breaking things because the credentials are JSON. Somehow the `%q` works but the `"%s"` doesn't.. I'm going to leave this as is for now (test code).